tests/unit/event_callback.c

```diff
@@ -0,0 +1,109 @@
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include <yatl/lite.h>
+
+#include <libdrizzle-5.1/libdrizzle.h>
+#include <libdrizzle-5.1/constants.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+extern drizzle_return_t drizzle_event_callback(drizzle_st *con, short events,
+  void *context);
+extern drizzle_return_t drizzle_event_callback(drizzle_st *con, short events,
+  void *context)
+{
+  ASSERT_TRUE((events != -1));
+  int* cxt_ptr = (int*) drizzle_context(con);
+  ASSERT_NOT_NULL_(cxt_ptr, "Drizzle connection context is null");
+  ASSERT_EQ_(1, cxt_ptr[0], "Invalid drizzle context: Got '%d', Expected '1'",
+    cxt_ptr[0]);
+
+  cxt_ptr = (int*)context;
+  ASSERT_NOT_NULL_(cxt_ptr, "Drizzle Event Callback context is null");
+  (*cxt_ptr)++;
+
+  return DRIZZLE_RETURN_OK;
+}
+
+int main(int argc, char *argv[])
+{
+  (void) argc;
+  (void) argv;
+
+  int cxt_a = 1;
+  int cxt_b = 0;
+
+  drizzle_st *con= drizzle_create(getenv("MYSQL_SERVER"),
+                                  getenv("MYSQL_PORT") ? atoi("MYSQL_PORT") :
+                                    DRIZZLE_DEFAULT_TCP_PORT,
+                                  getenv("MYSQL_USER"),
+                                  getenv("MYSQL_PASSWORD"),
+                                  getenv("MYSQL_SCHEMA"), 0);
+  ASSERT_NOT_NULL_(con, "Drizzle connection object creation error");
+
+  drizzle_set_context(con, (void*)&cxt_a);
+  ASSERT_NOT_NULL_(drizzle_context(con) , "Drizzle context is null");
+
+  drizzle_set_event_watch_fn(con, drizzle_event_callback, (void*)&cxt_b);
+
+  drizzle_return_t driz_ret= drizzle_connect(con);
+  SKIP_IF_(driz_ret == DRIZZLE_RETURN_COULD_NOT_CONNECT, "%s",
+    drizzle_strerror(driz_ret));
+  ASSERT_EQ_(DRIZZLE_RETURN_OK, driz_ret, "%s(%s)", drizzle_error(con),
+    drizzle_strerror(driz_ret));
+
+  driz_ret= drizzle_quit(con);
+  ASSERT_EQ_(DRIZZLE_RETURN_OK, driz_ret, "%s", drizzle_strerror(driz_ret));
+
+  // The drizzle_event_callback is invoked by drizzle_set_events and
+  // drizzle_set_revents with io changes.
+  // Internally the changes occur during the following steps:
+  //
+  // 1. init drizzle connection (drizzle_connect)
+  // 2. while connecting to server (drizzle_state_connecting)
+  // 3. performing server handshake (drizzle_state_handshake_client_write)
+  // 4. closing connection (drizzle_quit)
```

Comment:
> Should the above documentation should be here or at a different part of the test. 
The other unittests don't have a description of the test apart from the assert statement messages. 
I guess it could be moved to the file header